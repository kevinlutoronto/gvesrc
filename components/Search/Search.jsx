// @flow
import React, { Component } from 'react';
import { getDataAttrs } from '../../dataUtils';
import type { Data } from '../../types';
import style from './style.scss';

type Props = {|
  clearable?: boolean,
  clearIconUrl?: string,
  disabled?: boolean,
  iconUrl: string,
  instructionText?: string,
  onChange?: (event: SyntheticEvent<HTMLInputElement>) => void,
  onKeyDown?: (event: SyntheticKeyboardEvent<HTMLInputElement>) => void,
  onClear?: (event: SyntheticEvent<HTMLInputElement>) => void,
  placeholder?: string,
  value?: string,
  data?: Data,
  autoFocus?: boolean,
|};

type State = {|
  text: React$Node,
|};

class Search extends Component<Props, State> {
  static defaultProps = {
    placeholder: 'Search...',
    autoFocus: false,
  };

  constructor(props: Props) {
    super(props);
    this.state = {
      text: '',
    };
  }

  componentWillReceiveProps(nextProps: Props) {
    const { value } = nextProps;

    if (value !== this.props.value) {
      this.setState({ text: value });
    }
  }

  onChange = (event: SyntheticEvent<HTMLInputElement>): void => {
    const { onChange } = this.props;

    this.setState({ text: event.currentTarget.value });

    onChange && onChange(event);
  };

  onClear = (event: SyntheticEvent<HTMLInputElement>): void => {
    const { onClear } = this.props;

    this.setState({ text: '' });

    if (onClear) {
      onClear(event);
    }
  };

  render() {
    const {
      clearable,
      clearIconUrl,
      disabled,
      iconUrl,
      instructionText,
      placeholder,
      autoFocus,
    } = this.props;

    return (
      <div {...getDataAttrs(this.props.data)} className={style.container}>
        <div className={style.inputWrapper}>
          <span className={style.searchIconContainer}>
            <img src={iconUrl} role="presentation" />
          </span>
          <input
            className={style.input}
            disabled={disabled}
            onChange={this.onChange}
            onKeyDown={this.props.onKeyDown}
            placeholder={placeholder}
            type="text"
            value={this.state.text}
            autoFocus={autoFocus}
          />
          {clearable && clearIconUrl && this.state.text !== '' && (
            <img
              className={style.clearableIcon}
              onClick={this.onClear}
              src={clearIconUrl}
              role="presentation"
            />
          )}
        </div>
        {instructionText && <div className={style.instructionText}>{instructionText}</div>}
      </div>
    );
  }
}

Search.displayName = 'Plasma@Search';

export default Search;
