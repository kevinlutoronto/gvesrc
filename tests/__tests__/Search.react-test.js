/* eslint-disable */

import React from 'react';
import Search from '../../components/Search/Search';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

describe('Search Component', () => {
  test('render', () => {
    const component = renderer.create(<Search />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('onClear', () => {
    const onClear = jest.fn();
    const component = mount(
      <Search clearable clearIconUrl="http://yolo.com/icon.png" onClear={onClear} />
    );
    component.setState({ text: 'text' });
    component.find('.clearableIcon').simulate('click');
    expect(onClear).toHaveBeenCalledTimes(1);
  });

  test('onChange', () => {
    const onChange = jest.fn();
    const component = mount(<Search onChange={onChange} />);
    component.find('input').prop('onChange')({ currentTarget: { value: 'YOLO' } });
    expect(onChange).toHaveBeenCalledTimes(1);
    expect(component.state('text')).toBe('YOLO');
  });
});
