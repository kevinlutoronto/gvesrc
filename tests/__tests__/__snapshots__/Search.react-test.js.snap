// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component render 1`] = `
<div
  className="container"
>
  <div
    className="inputWrapper"
  >
    <span
      className="searchIconContainer"
    >
      <img
        role="presentation"
        src={undefined}
      />
    </span>
    <input
      autoFocus={false}
      className="input"
      disabled={undefined}
      onChange={[Function]}
      onKeyDown={undefined}
      placeholder="Search..."
      type="text"
      value=""
    />
  </div>
</div>
`;
